File: src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/ConcreteVisitor1.ts
import {ConcreteElementA} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementA";
import {ConcreteElementB} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementB";
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/Visitor";

class ConcreteVisitor1 implements Visitor {
    public visitConcreteElementA(element: ConcreteElementA): void {
        console.log(`${element.exclusiveMethodOfConcreteElementA()} from ConcreteVisitor1`);
    }

    public visitConcreteElementB(element: ConcreteElementB): void {
        console.log(`${element.specialMethodOfConcreteElementB()} from ConcreteVisitor1`);
    }
}

export { ConcreteVisitor1 };

File: java_patterns/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/Visitors/ConcreteVisitor1.java
package Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Visitors;

import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementA;
import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements.ConcreteElementB;

public class ConcreteVisitor1 implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println(element.exclusiveMethodOfConcreteElementA() + " from ConcreteVisitor1");
    }

    @Override
    public void visit(ConcreteElementB element) {
        System.out.println(element.specialMethodOfConcreteElementB() + " from ConcreteVisitor1");
    }
}
